Conflicting NPM dependencies

Trying to update the Flatpak's npm dependencies, I get this block of warnings about conflicting or unmet deps:
```
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-three/drei@9.122.0
npm warn Found: @react-three/fiber@9.1.0
npm warn node_modules/@react-three/fiber
npm warn   @react-three/fiber@"^9.1.0" from the root project
npm warn   4 more (@react-three/drei, @react-three/postprocessing, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @react-three/fiber@"^8" from @react-three/drei@9.122.0
npm warn node_modules/r3f-perf/node_modules/@react-three/drei
npm warn   @react-three/drei@"^9.103.0" from r3f-perf@7.2.3
npm warn   node_modules/r3f-perf
npm warn
npm warn Conflicting peer dependency: @react-three/fiber@8.18.0
npm warn node_modules/@react-three/fiber
npm warn   peer @react-three/fiber@"^8" from @react-three/drei@9.122.0
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei
npm warn     @react-three/drei@"^9.103.0" from r3f-perf@7.2.3
npm warn     node_modules/r3f-perf
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-three/drei@9.122.0
npm warn Found: react@19.1.0
npm warn node_modules/react
npm warn   react@"^19.1.0" from the root project
npm warn   22 more (@radix-ui/react-icons, @react-three/drei, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^18" from @react-three/drei@9.122.0
npm warn node_modules/r3f-perf/node_modules/@react-three/drei
npm warn   @react-three/drei@"^9.103.0" from r3f-perf@7.2.3
npm warn   node_modules/r3f-perf
npm warn
npm warn Conflicting peer dependency: react@18.3.1
npm warn node_modules/react
npm warn   peer react@"^18" from @react-three/drei@9.122.0
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei
npm warn     @react-three/drei@"^9.103.0" from r3f-perf@7.2.3
npm warn     node_modules/r3f-perf
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-three/drei@9.122.0
npm warn Found: react-dom@19.1.0
npm warn node_modules/react-dom
npm warn   react-dom@"^19.1.0" from the root project
npm warn   4 more (@react-three/drei, @react-three/fiber, r3f-perf, react-use-measure)
npm warn
npm warn Could not resolve dependency:
npm warn peerOptional react-dom@"^18" from @react-three/drei@9.122.0
npm warn node_modules/r3f-perf/node_modules/@react-three/drei
npm warn   @react-three/drei@"^9.103.0" from r3f-perf@7.2.3
npm warn   node_modules/r3f-perf
npm warn
npm warn Conflicting peer dependency: react-dom@18.3.1
npm warn node_modules/react-dom
npm warn   peerOptional react-dom@"^18" from @react-three/drei@9.122.0
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei
npm warn     @react-three/drei@"^9.103.0" from r3f-perf@7.2.3
npm warn     node_modules/r3f-perf
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-spring/three@9.7.5
npm warn Found: react@19.1.0
npm warn node_modules/react
npm warn   react@"^19.1.0" from the root project
npm warn   22 more (@radix-ui/react-icons, @react-three/drei, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/three@9.7.5
npm warn node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn   @react-spring/three@"~9.7.5" from @react-three/drei@9.122.0
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei
npm warn
npm warn Conflicting peer dependency: react@18.3.1
npm warn node_modules/react
npm warn   peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/three@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn     @react-spring/three@"~9.7.5" from @react-three/drei@9.122.0
npm warn     node_modules/r3f-perf/node_modules/@react-three/drei
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: react-composer@5.0.3
npm warn Found: react@19.1.0
npm warn node_modules/react
npm warn   react@"^19.1.0" from the root project
npm warn   22 more (@radix-ui/react-icons, @react-three/drei, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^15.0.0 || ^16.0.0 || ^17.0.0 || ^18.0.0" from react-composer@5.0.3
npm warn node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/react-composer
npm warn   react-composer@"^5.0.3" from @react-three/drei@9.122.0
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei
npm warn
npm warn Conflicting peer dependency: react@18.3.1
npm warn node_modules/react
npm warn   peer react@"^15.0.0 || ^16.0.0 || ^17.0.0 || ^18.0.0" from react-composer@5.0.3
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/react-composer
npm warn     react-composer@"^5.0.3" from @react-three/drei@9.122.0
npm warn     node_modules/r3f-perf/node_modules/@react-three/drei
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-spring/animated@9.7.5
npm warn Found: react@19.1.0
npm warn node_modules/react
npm warn   react@"^19.1.0" from the root project
npm warn   22 more (@radix-ui/react-icons, @react-three/drei, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/animated@9.7.5
npm warn node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three/node_modules/@react-spring/animated
npm warn   @react-spring/animated@"~9.7.5" from @react-spring/three@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn   1 more (@react-spring/core)
npm warn
npm warn Conflicting peer dependency: react@18.3.1
npm warn node_modules/react
npm warn   peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/animated@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three/node_modules/@react-spring/animated
npm warn     @react-spring/animated@"~9.7.5" from @react-spring/three@9.7.5
npm warn     node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn     1 more (@react-spring/core)
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-spring/core@9.7.5
npm warn Found: react@19.1.0
npm warn node_modules/react
npm warn   react@"^19.1.0" from the root project
npm warn   22 more (@radix-ui/react-icons, @react-three/drei, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/core@9.7.5
npm warn node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three/node_modules/@react-spring/core
npm warn   @react-spring/core@"~9.7.5" from @react-spring/three@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn
npm warn Conflicting peer dependency: react@18.3.1
npm warn node_modules/react
npm warn   peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/core@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three/node_modules/@react-spring/core
npm warn     @react-spring/core@"~9.7.5" from @react-spring/three@9.7.5
npm warn     node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-spring/shared@9.7.5
npm warn Found: react@19.1.0
npm warn node_modules/react
npm warn   react@"^19.1.0" from the root project
npm warn   22 more (@radix-ui/react-icons, @react-three/drei, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/shared@9.7.5
npm warn node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three/node_modules/@react-spring/shared
npm warn   @react-spring/shared@"~9.7.5" from @react-spring/three@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn   2 more (@react-spring/animated, @react-spring/core)
npm warn
npm warn Conflicting peer dependency: react@18.3.1
npm warn node_modules/react
npm warn   peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/shared@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three/node_modules/@react-spring/shared
npm warn     @react-spring/shared@"~9.7.5" from @react-spring/three@9.7.5
npm warn     node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn     2 more (@react-spring/animated, @react-spring/core)
```

This still fails.

```
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-three/drei@9.122.0
npm warn Found: @react-three/fiber@9.1.2
npm warn node_modules/@react-three/fiber
npm warn   @react-three/fiber@"^9.1.2" from the root project
npm warn   4 more (@react-three/drei, @react-three/postprocessing, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer @react-three/fiber@"^8" from @react-three/drei@9.122.0
npm warn node_modules/r3f-perf/node_modules/@react-three/drei
npm warn   @react-three/drei@"^9.103.0" from r3f-perf@7.2.3
npm warn   node_modules/r3f-perf
npm warn
npm warn Conflicting peer dependency: @react-three/fiber@8.18.0
npm warn node_modules/@react-three/fiber
npm warn   peer @react-three/fiber@"^8" from @react-three/drei@9.122.0
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei
npm warn     @react-three/drei@"^9.103.0" from r3f-perf@7.2.3
npm warn     node_modules/r3f-perf
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-three/drei@9.122.0
npm warn Found: react@19.1.0
npm warn node_modules/react
npm warn   react@"^19.1.0" from the root project
npm warn   22 more (@radix-ui/react-icons, @react-three/drei, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^18" from @react-three/drei@9.122.0
npm warn node_modules/r3f-perf/node_modules/@react-three/drei
npm warn   @react-three/drei@"^9.103.0" from r3f-perf@7.2.3
npm warn   node_modules/r3f-perf
npm warn
npm warn Conflicting peer dependency: react@18.3.1
npm warn node_modules/react
npm warn   peer react@"^18" from @react-three/drei@9.122.0
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei
npm warn     @react-three/drei@"^9.103.0" from r3f-perf@7.2.3
npm warn     node_modules/r3f-perf
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-three/drei@9.122.0
npm warn Found: react-dom@19.1.0
npm warn node_modules/react-dom
npm warn   react-dom@"^19.1.0" from the root project
npm warn   4 more (@react-three/drei, @react-three/fiber, r3f-perf, react-use-measure)
npm warn
npm warn Could not resolve dependency:
npm warn peerOptional react-dom@"^18" from @react-three/drei@9.122.0
npm warn node_modules/r3f-perf/node_modules/@react-three/drei
npm warn   @react-three/drei@"^9.103.0" from r3f-perf@7.2.3
npm warn   node_modules/r3f-perf
npm warn
npm warn Conflicting peer dependency: react-dom@18.3.1
npm warn node_modules/react-dom
npm warn   peerOptional react-dom@"^18" from @react-three/drei@9.122.0
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei
npm warn     @react-three/drei@"^9.103.0" from r3f-perf@7.2.3
npm warn     node_modules/r3f-perf
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-spring/three@9.7.5
npm warn Found: react@19.1.0
npm warn node_modules/react
npm warn   react@"^19.1.0" from the root project
npm warn   22 more (@radix-ui/react-icons, @react-three/drei, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/three@9.7.5
npm warn node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn   @react-spring/three@"~9.7.5" from @react-three/drei@9.122.0
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei
npm warn
npm warn Conflicting peer dependency: react@18.3.1
npm warn node_modules/react
npm warn   peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/three@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn     @react-spring/three@"~9.7.5" from @react-three/drei@9.122.0
npm warn     node_modules/r3f-perf/node_modules/@react-three/drei
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: react-composer@5.0.3
npm warn Found: react@19.1.0
npm warn node_modules/react
npm warn   react@"^19.1.0" from the root project
npm warn   22 more (@radix-ui/react-icons, @react-three/drei, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^15.0.0 || ^16.0.0 || ^17.0.0 || ^18.0.0" from react-composer@5.0.3
npm warn node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/react-composer
npm warn   react-composer@"^5.0.3" from @react-three/drei@9.122.0
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei
npm warn
npm warn Conflicting peer dependency: react@18.3.1
npm warn node_modules/react
npm warn   peer react@"^15.0.0 || ^16.0.0 || ^17.0.0 || ^18.0.0" from react-composer@5.0.3
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/react-composer
npm warn     react-composer@"^5.0.3" from @react-three/drei@9.122.0
npm warn     node_modules/r3f-perf/node_modules/@react-three/drei
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-spring/animated@9.7.5
npm warn Found: react@19.1.0
npm warn node_modules/react
npm warn   react@"^19.1.0" from the root project
npm warn   22 more (@radix-ui/react-icons, @react-three/drei, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/animated@9.7.5
npm warn node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three/node_modules/@react-spring/animated
npm warn   @react-spring/animated@"~9.7.5" from @react-spring/three@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn   1 more (@react-spring/core)
npm warn
npm warn Conflicting peer dependency: react@18.3.1
npm warn node_modules/react
npm warn   peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/animated@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three/node_modules/@react-spring/animated
npm warn     @react-spring/animated@"~9.7.5" from @react-spring/three@9.7.5
npm warn     node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn     1 more (@react-spring/core)
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-spring/core@9.7.5
npm warn Found: react@19.1.0
npm warn node_modules/react
npm warn   react@"^19.1.0" from the root project
npm warn   22 more (@radix-ui/react-icons, @react-three/drei, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/core@9.7.5
npm warn node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three/node_modules/@react-spring/core
npm warn   @react-spring/core@"~9.7.5" from @react-spring/three@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn
npm warn Conflicting peer dependency: react@18.3.1
npm warn node_modules/react
npm warn   peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/core@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three/node_modules/@react-spring/core
npm warn     @react-spring/core@"~9.7.5" from @react-spring/three@9.7.5
npm warn     node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @react-spring/shared@9.7.5
npm warn Found: react@19.1.0
npm warn node_modules/react
npm warn   react@"^19.1.0" from the root project
npm warn   22 more (@radix-ui/react-icons, @react-three/drei, ...)
npm warn
npm warn Could not resolve dependency:
npm warn peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/shared@9.7.5
npm warn node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three/node_modules/@react-spring/shared
npm warn   @react-spring/shared@"~9.7.5" from @react-spring/three@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn   2 more (@react-spring/animated, @react-spring/core)
npm warn
npm warn Conflicting peer dependency: react@18.3.1
npm warn node_modules/react
npm warn   peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @react-spring/shared@9.7.5
npm warn   node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three/node_modules/@react-spring/shared
npm warn     @react-spring/shared@"~9.7.5" from @react-spring/three@9.7.5
npm warn     node_modules/r3f-perf/node_modules/@react-three/drei/node_modules/@react-spring/three
npm warn     2 more (@react-spring/animated, @react-spring/core)
```

> Nothing jumps out as wrong comparing the new`node_sources.json` here: https://raw.githubusercontent.com/flathub/io.github.neil_morrison44.pocket-sync/76f927a9fa99e15b158df8d7e0b3690c3fa7c48c/node-sources.json

The problem is that the versions in the `node_sources.json` file aren't exact version, so if upstream updates the package in an incompatible way...

My guess is that if you nuke all your npm download caches and the `package-lock.json` file, you should run into the same problems.

--- a/package.json
+++ b/package.json
@@ -43,7 +43,6 @@
     "mdast-util-gfm-autolink-literal": "2.0.0",
     "postprocessing": "^6.37.8",
     "pretty-bytes": "^7.1.0",
-    "r3f-perf": "^7.2.3",
     "react": "^19.2.0",
     "react-dom": "^19.2.0",
     "react-i18next": "^16.2.0",

